Tests/DependencyInjection/DoctrineExtensionTest.php

```diff
@@ -927,6 +930,51 @@ public function testInvalidCacheConfiguration(): void
         $extension->load([$config], $container);
     }
 
+    public function testLegacyCacheConfigUsingServiceDefinedByApplication(): void
+    {
+        $this->expectNotToPerformAssertions();
+        (new class () extends TestKernel {
+            public function registerContainerConfiguration(LoaderInterface $loader): void
+            {
+                parent::registerContainerConfiguration($loader);
+                $loader->load(static function (ContainerBuilder $containerBuilder): void {
+                    $containerBuilder->loadFromExtension(
+                        'doctrine',
+                        ['orm' => ['query_cache_driver' => ['type' => 'service', 'id' => 'custom_cache_service']]]
+                    );
+                    $containerBuilder->setDefinition(
+                        'custom_cache_service',
+                        (new Definition(DoctrineProvider::class))
+                            ->setArguments([new Definition(ArrayAdapter::class)])
+                            ->setFactory([DoctrineProvider::class, 'wrap'])
+                    );
+                });
+            }
+        })->boot();
+    }
+
+    /** @group legacy */
```

Comment:
Rather than marking this legacy, change metadata_cache_driver to  query_cache_driver like was done in other method.